service/matching/matching_engine.go

```diff
@@ -2010,21 +2008,23 @@ func (e *matchingEngineImpl) updatePhysicalTaskQueueGauge(pm *physicalTaskQueueM
 	e.gaugeMetrics.loadedPhysicalTaskQueueCount[physicalTaskQueueParameters] += delta
 	loadedPhysicalTaskQueueCounter := e.gaugeMetrics.loadedPhysicalTaskQueueCount[physicalTaskQueueParameters]
```

Comment:
not really related to this PR but since it's near here:
Unlock() after this line instead of defer, so the lock doesn't cover the metrics stuff?